src/rustzcash.rs

```diff
@@ -77,6 +78,45 @@ fn read_le(from: &[u8]) -> FrRepr {
     f
 }
 
+/// Reads an FsRepr from [u8] of length 32
+/// This will panic (abort) if length provided is
+/// not correct
+fn read_fs(from: &[u8]) -> FsRepr {
+    assert_eq!(from.len(), 32);
+
+    let mut f = <<Bls12 as JubjubEngine>::Fs as PrimeField>::Repr::default();
+    f.read_be(from).expect("length is 32 bytes");
+
+    f.as_mut().reverse();
+    for b in f.as_mut() {
+        *b = swap_bits_u64(*b);
+    }
+
+    f
+}
+
+/// Writes an FsRepr to [u8] of length 32
+fn write_fs(mut f: FsRepr, to: &mut [u8]) {
+    assert_eq!(to.len(), 32);
+
+    f.as_mut().reverse();
+    for b in f.as_mut() {
+        *b = swap_bits_u64(*b);
+    }
+
+    f.write_be(to).expect("length is 32 bytes");
```

Comment:
Now that #12 has been merged, this function needs to be updated (to use `read_le` and not swap bits).